diff -rc php-5.3.9-orig/ext/pdo_oci/config.m4 php-5.3.9/ext/pdo_oci/config.m4
*** php-5.3.9-orig/ext/pdo_oci/config.m4	2012-01-19 14:17:09.157259399 +0600
--- php-5.3.9/ext/pdo_oci/config.m4	2012-01-19 14:25:10.443939562 +0600
***************
*** 74,84 ****
  
    if test "instantclient" = "`echo $PDO_OCI_DIR | cut -d, -f1`" ; then
      AC_CHECK_SIZEOF(long int, 4)
-     if test "$ac_cv_sizeof_long_int" = "4" ; then
-       PDO_OCI_CLIENT_DIR="client"
-     else
-       PDO_OCI_CLIENT_DIR="client64"
-     fi
      PDO_OCI_IC_PREFIX="`echo $PDO_OCI_DIR | cut -d, -f2`"
      PDO_OCI_IC_VERS="`echo $PDO_OCI_DIR | cut -d, -f3`"
      if test -n "$PDO_OCI_IC_VERS"; then
--- 74,79 ----
***************
*** 89,113 ****
        fi
      fi
      AC_MSG_CHECKING([for oci.h])
!     if test -f $PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR)
!     elif test -f $PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/include/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/include)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/include)
      elif test -f $PDO_OCI_IC_PREFIX/sdk/include/oci.h ; then
        PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/sdk/include)
        AC_MSG_RESULT($PDO_OCI_IC_PREFIX/sdk/include)
!     elif test -f $PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/include/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/include)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/include)
      else
        AC_MSG_ERROR([I'm too dumb to figure out where the include dir is in your Instant Client install])
      fi
!     if test -f "$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
!     PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib"
!     elif test -f "$PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
!       PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/lib"
      elif test -f "$PDO_OCI_IC_PREFIX/libclntsh.$SHLIB_SUFFIX_NAME" ; then
        PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX"
      else
--- 84,121 ----
        fi
      fi
      AC_MSG_CHECKING([for oci.h])
!     if test -f $PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/client/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/client)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/client)
!     elif test -f $PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/client64/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/$client64)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/include/oracle/$PDO_OCI_IC_VERS/$client64)
!     elif test -f $PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client/include/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client/include)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client/include)
!     elif test -f $PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client64/include/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client64/include)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client64/include)
      elif test -f $PDO_OCI_IC_PREFIX/sdk/include/oci.h ; then
        PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/sdk/include)
        AC_MSG_RESULT($PDO_OCI_IC_PREFIX/sdk/include)
!     elif test -f $PDO_OCI_IC_PREFIX/client/include/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/client/include)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/client/include)
!     elif test -f $PDO_OCI_IC_PREFIX/client64/include/oci.h ; then
!       PHP_ADD_INCLUDE($PDO_OCI_IC_PREFIX/client64/include)
!       AC_MSG_RESULT($PDO_OCI_IC_PREFIX/client64/include)
      else
        AC_MSG_ERROR([I'm too dumb to figure out where the include dir is in your Instant Client install])
      fi
!     if test -f "$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
!       PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client/lib"
!     elif test -f "$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client64/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
!       PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/client64/lib"
!     elif test -f "$PDO_OCI_IC_PREFIX/client64/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
!       PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/client64/lib"
!     elif test -f "$PDO_OCI_IC_PREFIX/client64/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
!       PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/client64/lib"
      elif test -f "$PDO_OCI_IC_PREFIX/libclntsh.$SHLIB_SUFFIX_NAME" ; then
        PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX"
      else
