*** php-5.3.5_ORIG/ext/pdo_oci/oci_statement.c	2010-01-03 14:23:27.000000000 +0500
--- php-5.3.5/ext/pdo_oci/oci_statement.c	2011-02-22 16:27:52.000000000 +0500
***************
*** 295,301 ****
  						P->oci_type = SQLT_CHR;
  						value_sz = param->max_value_len;
  						if (param->max_value_len == 0) {
! 							value_sz = 1332; /* maximum size before value is interpreted as a LONG value */
  						}
  
  				}
--- 295,301 ----
  						P->oci_type = SQLT_CHR;
  						value_sz = param->max_value_len;
  						if (param->max_value_len == 0) {
! 							value_sz = 4000; /* maximum size before value is interpreted as a LONG value */
  						}
  
  				}
***************
*** 572,578 ****
  				S->cols[colno].datalen = 1024;
  #endif
  			} else {
! 				S->cols[colno].datalen = col->maxlen;
  			}
  			if (dtype == SQLT_BIN) {
  				S->cols[colno].datalen *= 3;
--- 572,578 ----
  				S->cols[colno].datalen = 1024;
  #endif
  			} else {
! 				S->cols[colno].datalen = col->maxlen * 3;
  			}
  			if (dtype == SQLT_BIN) {
  				S->cols[colno].datalen *= 3;
