diff -ru pinba_engine-0.0.6.orig/configure.in pinba_engine-0.0.6/configure.in
--- pinba_engine-0.0.6.orig/configure.in	2012-05-17 23:56:37.456072695 +0600
+++ pinba_engine-0.0.6/configure.in	2012-05-17 23:44:36.000000000 +0600
@@ -192,22 +192,38 @@
 else
 
   if test "x$with_mysql" = "xyes"; then
-    AC_MSG_ERROR([sorry, I'm not that smart to guess where the MySQL sources are, please specify the path])
-  fi
+    with_mysql="/usr/include/mysql"
 
-  with_mysql=`dir_resolve "$with_mysql"`
+    HEADERS="my_dir.h plugin.h mysql.h mysql_version.h"
+    for file in $HEADERS; do
+      if ! test -r "$with_mysql/$file"; then
+        AC_MSG_ERROR([Failed to find required header file $file in $with_mysql])
+      fi
+    done
+    MYSQL_INC="-I$with_mysql -I$with_mysql/private"
+    MISSING_HEADER="$with_mysql/private/mysql_priv.h"
+	AC_DEFINE([MYSQL_PRIVATE_HEADERS], [1], [Whether we have MySQL 5.5])
+
+    AC_MSG_RESULT([$with_mysql])
+  else
+
+    with_mysql=`dir_resolve "$with_mysql"`
+
+    HEADERS="include/my_dir.h include/mysql/plugin.h include/mysql.h include/mysql_version.h"
+    for file in $HEADERS; do
+      if ! test -r "$with_mysql/$file"; then
+        AC_MSG_ERROR([Failed to find required header file $file in $with_mysql, check the path and make sure you've run './configure ..<options>.. && cd include && make' in MySQL sources dir])
+      fi
+    done
+    MYSQL_INC="-I$with_mysql/sql -I$with_mysql/include -I$with_mysql/regex -I$with_mysql"
+    MISSING_HEADER="$with_myslq/sql/mysql_priv.h"
 
-  HEADERS="include/my_dir.h include/mysql/plugin.h include/mysql.h include/mysql_version.h"
-  for file in $HEADERS; do
-    if ! test -r "$with_mysql/$file"; then
-      AC_MSG_ERROR([Failed to find required header file $file in $with_mysql, check the path and make sure you've run './configure ..<options>.. && cd include && make' in MySQL sources dir])
-    fi
-  done
+    AC_MSG_RESULT([$with_mysql])
 
-  AC_MSG_RESULT([$with_mysql])
+  fi
 
   CFLAGS_old="$CFLAGS"
-  CFLAGS="-I$with_mysql/include $CFLAGS"
+  CFLAGS="$MYSQL_INC $CFLAGS"
 
   AC_MSG_CHECKING([for MySQL version >= 5.1])
   AC_TRY_RUN([
@@ -227,15 +243,13 @@
   ]) 
 
   dnl check for 5.5 version with missing files and broken method signatures
-  MISSING_HEADER="sql/mysql_priv.h"
-  if ! test -r "$with_mysql/$MISSING_HEADER"; then
+  if ! test -r "$MISSING_HEADER"; then
 	  AC_DEFINE([MYSQL_VERSION_5_5], [1], [Whether we have MySQL 5.5])
   fi
 
   CFLAGS="$CFLAGS_old"
 
   AC_DEFINE([MYSQL_SRC], [1], [Source directory for MySQL])
-  MYSQL_INC="-I$with_mysql/sql -I$with_mysql/include -I$with_mysql/regex -I$with_mysql"
 fi
 
 
diff -ru pinba_engine-0.0.6.orig/src/config.h.in pinba_engine-0.0.6/src/config.h.in
--- pinba_engine-0.0.6.orig/src/config.h.in	2012-05-17 23:56:37.456072695 +0600
+++ pinba_engine-0.0.6/src/config.h.in	2012-05-17 23:43:52.000000000 +0600
@@ -52,6 +52,9 @@
 /* Source directory for MySQL */
 #undef MYSQL_SRC
 
+/* Private headers for MySQL */
+#undef MYSQL_PRIVATE_HEADERS
+
 /* Whether we have MySQL 5.5 */
 #undef MYSQL_VERSION_5_5
 
diff -ru pinba_engine-0.0.6.orig/src/ha_pinba.cc pinba_engine-0.0.6/src/ha_pinba.cc
--- pinba_engine-0.0.6.orig/src/ha_pinba.cc	2012-05-17 23:56:37.456072695 +0600
+++ pinba_engine-0.0.6/src/ha_pinba.cc	2012-05-17 23:54:48.000000000 +0600
@@ -21,17 +21,32 @@
 #include "pinba.h"
 
 #define MYSQL_SERVER 1
-#ifdef PINBA_ENGINE_MYSQL_VERSION_5_5
-# include <include/mysql_version.h>
-# include <sql/field.h>
-# include <sql/structs.h>
-# include <sql/handler.h>
+
+#ifdef PINBA_ENGINE_MYSQL_PRIVATE_HEADERS
+# ifdef PINBA_ENGINE_MYSQL_VERSION_5_5
+#  include <mysql_version.h>
+#  include <field.h>
+#  include <structs.h>
+#  include <handler.h>
+# else
+#  include <mysql_priv.h>
+# endif
+# include <my_dir.h>
+# include <plugin.h>
+# include <mysql.h>
 #else
-# include <mysql_priv.h>
+# ifdef PINBA_ENGINE_MYSQL_VERSION_5_5
+#  include <include/mysql_version.h>
+#  include <sql/field.h>
+#  include <sql/structs.h>
+#  include <sql/handler.h>
+# else
+#  include <mysql_priv.h>
+# endif
+# include <my_dir.h>
+# include <mysql/plugin.h>
+# include <mysql.h>
 #endif
-#include <my_dir.h>
-#include <mysql/plugin.h>
-#include <mysql.h>
 
 #include "ha_pinba.h"
 
