diff -cr gmrun-0.9.2.orig/src/main.cc gmrun-0.9.2/src/main.cc
*** gmrun-0.9.2.orig/src/main.cc	2011-04-21 21:31:06.291503418 +0600
--- gmrun-0.9.2/src/main.cc	2011-04-21 21:32:08.116995414 +0600
***************
*** 295,302 ****
        term = "xterm -e";
      }
      tmp = term;
!     tmp += " '";
!     tmp += command + "'";
    } else {
      if (!configuration.get_string("Terminal", term)) {
        tmp = "xterm";
--- 295,302 ----
        term = "xterm -e";
      }
      tmp = term;
!     tmp += " ";
!     tmp += command;
    } else {
      if (!configuration.get_string("Terminal", term)) {
        tmp = "xterm";
