diff -ur vsFTPd-3.0.2-ext.1.orig/tunables.c vsFTPd-3.0.2-ext.1/tunables.c
--- vsFTPd-3.0.2-ext.1.orig/tunables.c	2013-06-23 09:25:57.068727274 +0600
+++ vsFTPd-3.0.2-ext.1/tunables.c	2013-06-23 09:31:35.471688971 +0600
@@ -307,7 +307,7 @@
   /* -rw------- */
   tunable_chown_upload_mode = 0666;
 
-  install_str_setting("/usr/share/empty", &tunable_secure_chroot_dir);
+  install_str_setting("/usr/share/vsftpd/empty", &tunable_secure_chroot_dir);
   install_str_setting("ftp", &tunable_ftp_username);
   install_str_setting("root", &tunable_chown_username);
   install_str_setting("/var/log/xferlog", &tunable_xferlog_file);
