diff -ur vsFTPd-3.0.2-ext.1.orig/Makefile vsFTPd-3.0.2-ext.1/Makefile
--- vsFTPd-3.0.2-ext.1.orig/Makefile	2013-06-23 10:08:15.246059814 +0600
+++ vsFTPd-3.0.2-ext.1/Makefile	2013-06-23 10:08:35.686161161 +0600
@@ -4,7 +4,7 @@
 IFLAGS  = -idirafter dummyinc
 #CFLAGS = -g
 CFLAGS	=	-O2 -fPIE -fstack-protector --param=ssp-buffer-size=4 \
-	-Wall -W -Wshadow -Werror -Wformat-security \
+	-Wall -W -Wshadow -Wformat-security \
 	-D_FORTIFY_SOURCE=2 \
 	#-pedantic -Wconversion
 
