diff -ur vsFTPd-3.0.2-ext.1.orig/sysdeputil.c vsFTPd-3.0.2-ext.1/sysdeputil.c
--- vsFTPd-3.0.2-ext.1.orig/sysdeputil.c	2013-06-23 09:35:12.233463512 +0600
+++ vsFTPd-3.0.2-ext.1/sysdeputil.c	2013-06-23 09:37:55.914743702 +0600
@@ -186,8 +186,7 @@
 
 #if defined(VSF_SYSDEP_HAVE_CAPABILITIES) && !defined(VSF_SYSDEP_HAVE_LIBCAP)
 #include <linux/capability.h>
-#include <errno.h>
-#include <syscall.h>
+#include <sys/syscall.h>
 int capset(cap_user_header_t header, const cap_user_data_t data)
 {
   return syscall(__NR_capset, header, data);
