diff -ur vsFTPd-3.0.2-ext.1.orig/twoprocess.c vsFTPd-3.0.2-ext.1/twoprocess.c
--- vsFTPd-3.0.2-ext.1.orig/twoprocess.c	2013-06-23 09:23:07.607151180 +0600
+++ vsFTPd-3.0.2-ext.1/twoprocess.c	2013-06-23 09:23:50.037551854 +0600
@@ -307,6 +307,7 @@
 {
   enum EVSFPrivopLoginResult e_login_result = kVSFLoginNull;
   char cmd;
+  vsf_sysutil_install_null_sighandler(kVSFSysUtilSigCHLD);
   /* Blocks */
   cmd = priv_sock_get_cmd(p_sess->parent_fd);
   if (cmd != PRIV_SOCK_LOGIN)
@@ -395,7 +396,6 @@
   int was_anon = anon;
   const struct mystr* p_orig_user_str = p_user_str;
   int newpid;
-  vsf_sysutil_install_null_sighandler(kVSFSysUtilSigCHLD);
   /* Tells the pre-login child all is OK (it may exit in response) */
   priv_sock_send_result(p_sess->parent_fd, PRIV_SOCK_RESULT_OK);
   if (!p_sess->control_use_ssl)
