http://bugs.gentoo.org/327443

--- install.sh
+++ install.sh
@@ -179,20 +179,6 @@
   exit 23
 fi
 
-
-# Install pm-utils hook only if pm-utils config dir is present, or force the
-# install if we have a DESTDIR.
-if [ -n "$DESTDIR" -o -d "$DESTDIR/usr/lib/pm-utils/sleep.d" ] ; then
-	if ( ! $INSTALL -D -m 755 usr/lib/pm-utils/sleep.d/01laptop-mode "$DESTDIR/usr/lib/pm-utils/sleep.d/01laptop-mode" ) ; then
-		echo "$0: Failed to install 01-laptop-mode into /usr/lib/pm-utils/sleep.d. Installation failed."
-		exit 36
-	fi
-fi
-
-if [ -f "$DESTDIR/usr/lib/pm-utils/sleep.d/99laptop-mode" ]; then
-	rm -f $DESTDIR/usr/lib/pm-utils/sleep.d/99laptop-mode;
-fi
-
 # udev rule
  if ( ! $INSTALL -D -m 644 etc/rules/99-laptop-mode.rules "$DESTDIR/etc/udev/rules.d/99-laptop-mode.rules" ) ; then
         echo "$0: Failed to install udev rule into /etc/udev/rules.d/ Installation failed."
 
