diff -ur oblogout-0.2.6.orig/src/logout.c oblogout-0.2.6/src/logout.c
--- oblogout-0.2.6.orig/src/logout.c	2012-05-14 23:14:36.832203901 +0600
+++ oblogout-0.2.6/src/logout.c	2012-05-14 23:09:25.000000000 +0600
@@ -293,7 +293,7 @@
 	/* нехрена проверять, отдавать и всё. коли ставиш, знаеш зачем */
 	sb_available_actions();
 	
-	if( G_LIKELY( logout_action & LOGOUT_ACTION_SUSPEND ) )
+	if( G_LIKELY( logout_action & LOGOUT_ACTION_SUSPEND ) && G_LIKELY( cfg.enable_suspend ) )
 	{
 		/* Create button */
 		button = sb_create_button("logout", _("_Suspend"), "suspend.png", LOGOUT_ACTION_SUSPEND);
@@ -301,7 +301,7 @@
 		gtk_box_pack_start(GTK_BOX(hbuttonbox), button, FALSE, FALSE, 0);	
 	}
 
-	if( G_LIKELY( logout_action & LOGOUT_ACTION_HIBERNATE ) )
+	if( G_LIKELY( logout_action & LOGOUT_ACTION_HIBERNATE ) && G_LIKELY( cfg.enable_hibernate ) )
 	{
 		button = sb_create_button("logout", _("_Hibernate"), "hibernate.png", LOGOUT_ACTION_HIBERNATE);
 		gtk_box_pack_start(GTK_BOX(hbuttonbox), button, FALSE, FALSE, 0);
diff -ur oblogout-0.2.6.orig/src/setting.c oblogout-0.2.6/src/setting.c
--- oblogout-0.2.6.orig/src/setting.c	2012-05-14 23:14:36.832203901 +0600
+++ oblogout-0.2.6/src/setting.c	2012-05-14 23:04:56.000000000 +0600
@@ -57,6 +57,8 @@
 	g_key_file_load_from_file(file, path, G_KEY_FILE_KEEP_COMMENTS, NULL);
 
 	g_key_file_set_string(file, "general", "theme", cfg.theme);
+	g_key_file_set_string(file, "general", "enable_suspend", cfg.enable_suspend);
+	g_key_file_set_string(file, "general", "enable_hibernate", cfg.enable_hibernate);
 	
 	g_key_file_set_integer(file, "button", "size", cfg.buttonsize);
 	
@@ -80,6 +82,8 @@
 config_load_from_keyfile(GKeyFile *file)
 {
 	cfg.theme = g_key_file_get_string(file, "general", "theme", NULL);
+	cfg.enable_suspend = g_key_file_get_string(file, "general", "enable_suspend", 0);
+	cfg.enable_hibernate = g_key_file_get_string(file, "general", "enable_hibernate", 0);
 	
 	cfg.buttonsize = (unsigned int) g_key_file_get_integer(file, "button", "size", NULL);
 
diff -ur oblogout-0.2.6.orig/src/setting.h oblogout-0.2.6/src/setting.h
--- oblogout-0.2.6.orig/src/setting.h	2012-05-14 23:14:36.832203901 +0600
+++ oblogout-0.2.6/src/setting.h	2012-05-14 23:00:18.000000000 +0600
@@ -36,6 +36,8 @@
 	gchar *labelcolor;
 	gchar *label_font;
 	GdkColor color_label;
+	guint8 enable_suspend;
+	guint8 enable_hibernate;
 } Config;
 
 extern Config cfg;
