diff -c oblogout-0.2.5.orig/Makefile.am oblogout-0.2.5/Makefile.am
*** oblogout-0.2.5.orig/Makefile.am	2011-10-21 08:36:51.847678213 +0600
--- oblogout-0.2.5/Makefile.am	2011-10-21 08:37:29.957679196 +0600
***************
*** 4,10 ****
  SUBDIRS = src po \
  	themes
  
! oblogoutdocdir = ${prefix}/doc/oblogout
  oblogoutdoc_DATA = \
  	README\
  	COPYING\
--- 4,10 ----
  SUBDIRS = src po \
  	themes
  
! oblogoutdocdir = ${prefix}/share/doc/oblogout
  oblogoutdoc_DATA = \
  	README\
  	COPYING\
